ccc-server/src/main.rs

```diff
@@ -62,6 +87,25 @@ async fn error_handler(error: BoxError) -> impl IntoResponse {
 	}
 }
 
+fn init_tracing(tracing: LogStructure) {
+	if tracing == LogStructure::Default {
+		tracing_subscriber::fmt::init();
+	} else {
+		let output: Box<dyn Layer<tracing_subscriber::Registry> + Send + Sync> = match tracing {
+			LogStructure::Default => unimplemented!(),
+			LogStructure::Debug => layer().fmt_fields(format::Pretty::default()).boxed(),
+			LogStructure::Json => layer().json().boxed(),
+			LogStructure::Pretty => layer()
+				.event_format(format::Format::default().with_source_location(false))
+				.fmt_fields(format::PrettyFields::new())
+				.with_target(false)
+				.boxed(),
+		};
+
+		tracing_subscriber::registry().with(output).init();
+	}
```

Comment:
You're handling the `LogStructure::Default` case in the `if`-case and in the `else`-case with just `unimplemented!()`.

Some ideas:

- The right macro, I think, is `unreachable!()` instead of `unimplemented!()` if we want to keep this structure; technically `Default` _is_ implemented, just not in the `else` control flow path.
- Arguably, the `registry()` is appropriate in all cases, even the Default, `fmt::init()` case. We want to default to the `fmt` subscriber, but having it receive spans from the registry is a fine tradeoff. Can we construct a `fmt` subscriber (using just the defaults) and wrap it? this way we shouldn't need to switch based on `Default`. The runtime cost is probably miniscule to do it this way, and the behavior gets less complex if we do it that way.
- `LogStructure` is passed by value, but not _consumed_ in the function body. We're passing the value and passing off ownership. We should either pass it by reference, or allow it to be copied. I would argue the latter is a much better idea — can we mark `LogStructure` as `Copy`?

Reply:
The first and third points make sense to me. I'm mostly grasping your second point.

You're right that the registry is still appropriate in all cases but I would have to check the docs to see if we can still achieve the same concise output with the registry call (if the Default case were to be kept). From what I was observing, the registry was driving all of the additional context and verbosity behind the span traces.

Could you help me understand how we want to change this then with those output goals in-mind?

Reply:
I think what I was suggesting was something along the lines of

```suggestion
	let output = match tracing {
		LogStructure::Default => layer().boxed(), // NOTE: Is this truly equivalent to fmt().init();?
		LogStructure::Debug => layer().fmt_fields(format::Pretty::default()).boxed(),
		LogStructure::Json => layer().json().boxed(),
		LogStructure::Pretty => layer()
			.event_format(format::Format::default().with_source_location(false))
			.fmt_fields(format::PrettyFields::new())
			.with_target(false)
			.boxed(),
	};

	tracing_subscriber::registry().with(output).init();
```

Reply:
Ah got it. I gave that a try initially and it did not return today's span experience. That's the only reason I went down the path of separating the two.